### PR TITLE
Add option to validate semaphore proof with root age

### PR DESCRIPTION
**Pull request by Dzejkop**
## PR Checklist

- [x] Added Tests
- [x] Added Documentation

### CODE REVIEW

**Review by dcbuild3r — COMMENTED** [review]
lgtm

File: src/app.rs [hunk 1]
```diff
@@ -461,6 +468,52 @@ impl App {
...
+            }
+            _ => (),
+        }
+
+        // Latest processed root is always valid
+        if root_state.root == processed_root {
+            return Ok(());
+        }
+
+        let max_root_age = Duration::seconds(max_root_age_seconds);
```
Comment (0xKitsune) [inline src/app.rs:496]: Can this be passed in as `Duration`?

File: src/app.rs [hunk 2]
```diff
@@ -461,6 +468,52 @@ impl App {
...
+
+        let mined_at = root_state.mined_valid_as_of;
+        let pending_as_of = root_state.pending_valid_as_of;
+
+        let now = chrono::Utc::now();
+
```
Comment (0xKitsune) [inline src/app.rs:502]: Do you need to initialize varibles here? Or can you just simplify to the following:

```rust
let root_age = if root_state.status == Status::Pending {
    chrono::Utc::now() - root_state.pending_valid_as_of;
} else {
    let mined_at = root_state.mined_valid_as_of.ok_or(ServerError::InvalidRoot)?;
    chrono::Utc::now() - mined_at
};
```

File: src/app.rs (hunk 1 above)
Comment (0xKitsune) [inline src/app.rs:496]: Can this be passed in as `Duration`?

File: src/app.rs (hunk 2 above)
Comment (Dzejkop) [reply src/app.rs:502]: Fair, but I'll leave the `now` variable.

**Review by 0xKitsune — APPROVED** [review]

**Pull request merged**